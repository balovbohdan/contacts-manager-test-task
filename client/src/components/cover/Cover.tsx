import * as React from 'react';
import {Link} from 'react-router-dom';

import {paths} from '@routes';

import * as css from './Cover.css';

export const Cover = () =>
    <div className={css.main}>
        <Img/>
        <Slogan/>
        <TryItBtn/>
    </div>;

const Img = () =>
    <div className={css.img}/>;

const Slogan = () =>
    <h1 className={css.slogan}>
        <span>CM</span> â€” awesome <span>C</span>ontacts <span>M</span>anager
    </h1>;

const TryItBtn = () =>
    <Link to={paths.contacts} className={css.tryItBtn}>
        <h3 className={css.tryItBtn_txt}>Try it!</h3>
    </Link>;