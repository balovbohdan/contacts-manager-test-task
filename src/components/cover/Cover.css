.main {
    width: 100%;
    height: calc(100vh - var(--header-height));

    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    background-color: var(--color-light-blue);
}

.img {
    margin-top: 20px;

    max-height: 400px;
    height: calc(100% - 50px);

    max-width: 500px;
    width: calc(100% - 20px);

    background-size: contain;
    background-repeat: no-repeat;
    background-position: center center;
    background-image: url(~@img/cover.svg);
}

.slogan {
    margin: 20px 20px 0;

    text-align: center;
    color: var(--color-white);
    font-family: var(--font-marker);
}

.slogan > span {
    display: inline;

    color: var(--color-yellow);
    font-family: var(--font-marker);
}

.tryItBtn {
    cursor: pointer;
    border-radius: 5px;
    background: rgba(255, 255, 255, .1);

    margin: 20px 0;
    padding: 10px 20px;
}

.tryItBtn .tryItBtn_txt {
    color: var(--color-red);
    font-family: var(--font-marker) !important;
}

.tryItBtn:hover {
    transition: var(--trans);
    background: rgba(255, 255, 255, .3);
}