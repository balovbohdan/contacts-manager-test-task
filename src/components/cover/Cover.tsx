import * as React from 'react';

import css from './Cover.css';

export const Cover = () =>
    <div className={css.main}>
        <Img/>
        <Slogan/>
        <TryItBtn/>
    </div>;

const Img = () =>
    <div className={css.img}/>;

const Slogan = () =>
    <h1 className={css.slogan}>
        <span>CM</span> â€” awesome <span>C</span>ontacts <span>M</span>anager
    </h1>;

const TryItBtn = () =>
    <div className={css.tryItBtn}>
        <h3 className={css.tryItBtn_txt}>Try it!</h3>
    </div>;