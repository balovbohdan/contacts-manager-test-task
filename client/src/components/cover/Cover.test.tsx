import {shallow} from 'enzyme';
import * as React from 'react';

import {Cover} from './Cover';

describe('Main page Cover', () => {
    it('Should match snapshot', () => {
        const cover = shallow(<Cover/>);

        expect(cover).toMatchSnapshot();
    });
});