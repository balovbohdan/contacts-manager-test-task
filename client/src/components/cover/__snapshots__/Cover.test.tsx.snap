// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page Cover Should match snapshot 1`] = `
<div
  className="main"
>
  <Img />
  <Slogan />
  <TryItBtn />
</div>
`;
